Makefile

```diff
@@ -1,11 +1,56 @@
-BUILD_DIR:=./bin
+BUILDFLAGS ?=
+unexport GOFLAGS
 
-build: clean
-	mkdir -p ${BUILD_DIR}
-	go build -o "${BUILD_DIR}/backplane-tools" main.go
+BASE_DIR=$(shell pwd)
+BIN_DIR=${BASE_DIR}/bin
 
-clean:
-	rm -rf "${BUILD_DIR}"
+all: format mod build test lint
 
+format: vet mod fmt mockgen ci-build docs
+
+fmt:
+	@echo "gofmt"
+	@gofmt -w -s .
+	@git diff --exit-code .
```

Comment:
**suggestion:** The `git diff --exit-code` here will result in a `make format` or `make all` target failing midway, just because the changes aren't commited. I always thought this is very annoying in `osdctl`, so would rather not spread it to other projects.